xbmc/peripherals/bus/PeripheralBus.h

```diff
@@ -54,7 +54,7 @@ namespace PERIPHERALS
     /*!
      * @return True if this bus needs to be polled for changes, false if this bus performs updates via callbacks
      */
-    bool NeedsPolling(void) const { return m_bNeedsPolling; }
+    bool NeedsPolling(void) const { CSingleLock lock(m_critSection); return m_bNeedsPolling; }
```

Comment:
@Paxxi What I mean is, you still need synchronization (or cas) where just an atomic doesn't cut it.
Here's the alternative that will just replacing them with atomic: https://github.com/tamland/xbmc/commit/420a04a41150ffefaf2223004a31b59c68ccf6b2
And you still need to lock over the atomics because of race conditions: https://github.com/tamland/xbmc/blob/420a04a41150ffefaf2223004a31b59c68ccf6b2/xbmc/peripherals/bus/PeripheralBus.cpp#L226 Possibly there are more similar issues.
Hench I think it's easier and safer to just use mutexes everywhere unless you know what you're doing and design for it from the beginning. I only wanted to fix some quick obvious issues here:)